cmd/commands/subscription_display.go

```diff
@@ -0,0 +1,97 @@
+package commands
+
+import (
+	"fmt"
+	"github.com/knative/eventing/pkg/apis/channels/v1alpha1"
+	"io"
+)
+
+type subscriptionDisplay struct {
+	headers       *[]subscriptionHeader
+	subscriptions *[]v1alpha1.Subscription
+}
+
+type stringExtractor func(*v1alpha1.Subscription) string
+
+type subscriptionHeader struct {
+	namedExtractor
+	padding int
```

Comment:
More a width than a padding. Also, it does not really belong to the header, but to the whole column.

This is why it may be best to just extract the value, then do a full pass to compute the widths, and then print.
Even if that means less "object oriented code"...